### PR TITLE
Feature/ic2 21 update prepstatus

### PR DESCRIPTION
**Pull request by dhlee-icon**
Please merge to icondev branch
- update base.go to update prep status, VTotal, VFail, VFailCont
- add base_test.go
- edit icstate.prepstatus.go to add VFailCont() method

### CODE REVIEW

File: icon/iiss/base.go [hunk 1]
```diff
@@ -178,8 +179,42 @@ func handleConsensusInfo(wc state.WorldContext) error {
...
+	}
+	err := handlePrepStatus(es.State, prepAddressList, voted)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func handlePrepStatus(state *icstate.State, prepAddressList []module.Address, voted []bool) error {
+	if len(prepAddressList) != 0 {
```
Comment (eunsoo-icon) [inline icon/iiss/base.go:200]: You can delete this if statement for optimization

Comment (dhlee-icon) [reply icon/iiss/base.go:200]: If you mean that combine two methods into one, I intended to seperate those to make base_test.go

Comment (eunsoo-icon) [reply icon/iiss/base.go:200]: No. I talked about 'if' statement.
```go
if len(prepAddressList) != 0 {
		for i := 0; i < len(prepAddressList); i += 1 {
```
If prepAddressList has no value, for loop will not do anything.

Therefore, 'if' statement is meaningless

File: icon/iiss/base.go [hunk 2]
```diff
@@ -178,8 +179,42 @@ func handleConsensusInfo(wc state.WorldContext) error {
 		return err
 	}
 
-	// TODO update P-Rep status
+	// update P-rep status, vtotal, vfail, vfailcont
+	proposerExist := false
+	for _, p := range prepAddressList {
+		if p == proposer {
```
Comment (mksong-icon) [inline icon/iiss/base.go:185]: Please use p.Equal() instead of simple "==" operator.

Comment (dhlee-icon) [reply icon/iiss/base.go]: I reflected it, Thanks.

File: icon/iiss/base.go (hunk 1 above)
Comment (dhlee-icon) [reply icon/iiss/base.go]: understood, I reflected it, thank you

File: icon/iiss/base.go [hunk 3]
```diff
@@ -155,20 +156,20 @@ func handleConsensusInfo(wc state.WorldContext) error {
...
+	proposer := es.pm.GetPRepByNode(csi.Proposer()).Owner()
 	validators := csi.Voters()
 	voted := csi.Voted()
 	voters := make([]module.Address, 0)
-
+	prepAddressList := make([]module.Address, 0)
 	if validators != nil {
 		for i := 0; i < validators.Len(); i += 1 {
+			v, _ := validators.Get(i)
+			prepAddressList = append(prepAddressList, es.pm.GetPRepByNode(v.Address()).Owner())
```
Comment (eunsoo-icon) [inline icon/iiss/base.go:167]:
```suggestion
			owner := es.pm.GetPRepByNode(v.Address()).Owner()
			prepAddressList = append(prepAddressList, owner)
```

File: icon/iiss/base.go [hunk 4]
```diff
@@ -155,20 +156,20 @@ func handleConsensusInfo(wc state.WorldContext) error {
...
 			if voted[i] {
-				v, _ := validators.Get(i)
-				voters = append(voters, v.Address())
+				voters = append(voters, es.pm.GetPRepByNode(v.Address()).Owner())
```
Comment (eunsoo-icon) [inline icon/iiss/base.go:169]:
```suggestion
				voters = append(voters, owner)
```

**Review by eunsoo-icon — CHANGES_REQUESTED** [review]

**Review by eunsoo-icon — APPROVED** [review]

**Pull request merged**